source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,33 @@
 class UrlsController < ApplicationController
+  #GET urls
+  #def index
+    #@urls = Url.all #rake db:reset db:migrate
+  #end
+
+  #GET url
+  def new
+    @url = Url.new
+    @urls = Url.all
+  end
+
+  #POST urls
+  def create
+    check = Url.find_by(:address => params[:url][:address])
+    if !check
+      @url = Url.create(:address => params[:url][:address])
+    end
+    flash[:error] = @url.errors.full_messages.to_sentence
+    redirect_to root_path
+  end
+
+  def move
+    @url = Url.where :bitly_clone => params[:short]
```

Comment:
`Url.find_by(:bitly_clone => params[:short])` gives you either the found record or `nil`. That simplifies the later code a little, since it isn't wrapped in an array.